### PR TITLE
feat: added whomai cmd to coder cli

### PR DESCRIPTION
**Pull request by BRAVO68WEB**
Example Usage:-
image

### CODE REVIEW

File: cli/whoami.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		Middleware: serpent.Chain(
+			serpent.RequireNArgs(0),
+			r.InitClient(client),
+		),
+		Handler: func(inv *serpent.Invocation) error {
+			ctx := inv.Context()
+			// Fetch the user info
+			resp, err := client.User(ctx, codersdk.Me)
+			// Get Coder instance url
+			deployConfig, _ := client.DeploymentConfig(ctx)
```
Comment (Emyrk) [inline cli/whoami.go:27]: Not all users can query this endpoint. If we just want the url though, we can use `client.URL` instead.

Comment (BRAVO68WEB) [reply cli/whoami.go:27]: Ohh, Ok I didn't know that

**Comment by BRAVO68WEB** [comment]
Added Requested changes @Emyrk

**Comment by Emyrk** [comment]
@BRAVO68WEB happy to include this. 

Can you add a test for both when this works and when the user is logged out?

You will also need to run a `make gen` to generate the docs associated with the cli.

I can assist in any of these comments :+1:

**Comment by BRAVO68WEB** [comment]
Yeah about the tests, I found writing test for this project hard

**Comment by Emyrk** [comment]
For testing the failure condition, you can copy this: https://github.com/coder/coder/blob/44cb400c8e5b8c2f1a428f75210a56066015f3e1/cli/login_test.go#L26-L33

For the successful case, this one has the setup you would need: https://github.com/coder/coder/blob/44cb400c8e5b8c2f1a428f75210a56066015f3e1/cli/publickey_test.go#L15-L27

Essentially just a test to verify an error when not logged in, and no error when the user is.

**Review by Emyrk — APPROVED** [review]

**Comment by Emyrk** [comment]
Ah, you have to run `make update-golden-files -B` as well. (I add `-B` to be sure).
@BRAVO68WEB

**Review by Emyrk — CHANGES_REQUESTED** [review]
Will approve and merge when CI passes

**Comment by BRAVO68WEB** [comment]
> Ah, you have to run `make update-golden-files -B` as well. (I add `-B` to be sure). @BRAVO68WEB

Done 👍🏻

**Comment by BRAVO68WEB** [comment]
BTW I got this, is it expected ?
image

**Comment by Emyrk** [comment]
> BTW I got this, is it expected ? image

That can happen based on some helm state. It is something we should improve in our makefile.

Regardless, it looks like you got the right files generated :+1:. I approved CI, let's see what it looks like.

**Comment by BRAVO68WEB** [comment]
CI Passed 😄

**Review by Emyrk — APPROVED** [review]

**Comment by Emyrk** [comment]
@BRAVO68WEB merged! Thanks for the contribution!

**Pull request merged**